tests/unit/ReportTitleUtilsTest.ts

```diff
@@ -0,0 +1,216 @@
+            const result = removeTitleFieldFromReport(mockReportID);
+
+            expect(result).toHaveLength(1);
+            expect(result.at(0)).toEqual({
+                onyxMethod: Onyx.METHOD.MERGE,
+                key: `${ONYXKEYS.COLLECTION.REPORT_NAME_VALUE_PAIRS}${mockReportID}`,
+                value: {
+                    // eslint-disable-next-line @typescript-eslint/naming-convention
+                    expensify_text_title: null,
+                },
+            });
+        });
+    });
+
+    describe('shouldUpdateTitleField', () => {
+        const mockReportID = '12345';
+        const mockTitleField = {
+            defaultValue: 'Test Title',
+        };
+
+        beforeEach(() => {
+            mockedReportUtils.getReportNameValuePairs.mockReturnValue({
+                // eslint-disable-next-line @typescript-eslint/naming-convention
+                expensify_text_title: mockTitleField,
+            });
+        });
+
+        it('should return false when report is chat report', () => {
+            const mockReport: Report = {
+                reportID: mockReportID,
+                type: CONST.REPORT.TYPE.CHAT,
+            } as Report;
+
+            mockedReportUtils.isChatReport.mockReturnValue(true);
+
+            const result = shouldUpdateTitleField(mockReport);
+
+            expect(mockedReportUtils.isChatReport).toHaveBeenCalledWith(mockReport);
+            expect(result).toBe(false);
+        });
+
+        it('should return false when report has no title field in RNVP', () => {
+            const mockReport: Report = {
+                reportID: mockReportID,
+                type: CONST.REPORT.TYPE.EXPENSE,
+            } as Report;
+
+            mockedReportUtils.isChatReport.mockReturnValue(false);
+            mockedReportUtils.getReportNameValuePairs.mockReturnValue({});
+
+            const result = shouldUpdateTitleField(mockReport);
+
+            expect(result).toBe(false);
+        });
+
+        it('should return true when non-chat report has title field', () => {
+            const mockReport: Report = {
+                reportID: mockReportID,
+                type: CONST.REPORT.TYPE.EXPENSE,
+            } as Report;
+
+            mockedReportUtils.isChatReport.mockReturnValue(false);
+
+            const result = shouldUpdateTitleField(mockReport);
+
+            expect(mockedReportUtils.isChatReport).toHaveBeenCalledWith(mockReport);
+            expect(result).toBe(true);
+        });
+
+        it('should return true for IOU report with title field', () => {
+            const mockReport: Report = {
+                reportID: mockReportID,
+                type: CONST.REPORT.TYPE.IOU,
+            } as Report;
+
+            mockedReportUtils.isChatReport.mockReturnValue(false);
+
+            const result = shouldUpdateTitleField(mockReport);
+
+            expect(result).toBe(true);
+        });
```

Comment:
If we are going to test this behavior, then it should match the actual implementation we want, which is that IOU reports should not be updated.